File: Api/AvaluxAuth.DataAccess/Entities/ProviderEntity.cs
using System.ComponentModel.DataAnnotations;

namespace AvaluxAuth.DataAccess.Entities;

internal class ProviderEntity
{
    public Guid Id { get; init; }
    public Guid ApplicationId { get; init; }
    public int ProviderId { get; init; }

    [MaxLength(100)] public string? ClientId { get; init; }
    [MaxLength(100)] public string? ClientSecret { get; init; }
    public bool SaveTokens { get; init; }
    public string[] DefaultScope { get; init; } = [];

    public required DateTime CreatedAt { get; init; }
    public DateTime? DeletedAt { get; init; }

    public ApplicationEntity Application { get; init; } = null!;
    public ICollection<AccountEntity> Accounts { get; init; } = [];
}

File: UI/src/app/entities/provider-entity.ts
import {Moment} from 'moment';

export interface ProviderEntity {
  id: string;
  providerId: number;
  applicationId: string;
  parameters: ProviderParametersEntity;
  createdAt: Moment;
  deletedAt: Moment | null;
}

export interface ProviderInfoEntity {
  name: string;
  key: string;
  id: number;
  fields: string[];
  url?: string;
}

export interface ProviderParametersEntity {
  clientName?: string;
  clientId?: string;
  clientSecret?: string;
  providerUrl?: string;
  saveTokens: boolean;
  defaultScope: string[];
}
